@testable import ExampleAppBinary doesn't work

I've mentioned this issue on Slack channel, and wanted to try it with AirBnB's custom fork of `buck`.

Still no luck though.

Adding `@testable import ExampleAppBinary` to any of the tests, e.g. to `BuckSampleTests.swift` or `BuckSampleUITests.swift` causes this error:

```
Command failed with exit code 1.
stderr: App/UITests/BuckSampleUITests.swift:10:18: error: no such module 'ExampleAppBinary'
@testable import ExampleAppBinary
                 ^
```

While I can definitely find the `ExampleAppBinary` module with `find buck-out -name "*ExampleApp*"`

```
buck-out/gen/App/ExampleAppBinary#iphonesimulator-x86_64,swift-compile
buck-out/gen/App/ExampleAppBinary#iphonesimulator-x86_64,swift-compile/ExampleAppBinary.swiftmodule
buck-out/gen/App/ExampleAppBinary#iphonesimulator-x86_64,swift-compile/ExampleAppBinary-Swift.h
buck-out/gen/App/ExampleAppBinary#iphonesimulator-x86_64,swift-compile/ExampleAppBinary.o
```

I can't use it in Unit or UI tests though.

The original issue is here: https://github.com/facebook/buck/issues/2074

P.S.
I know it's open source and all that, but I don't have the time and code base knowledge to locate the change that removed this functionality and revert just the bits that make app module imports work again.

Since this is the only up-to-date buck demo of mix-n-match project there is (🥇), I thought raising it here would draw some attention to it.

Also, I though most of AirBnB's changes made it to master already... At least I was able to use `import` statements in our mix-n-match project with `modular=True` flag using current `master`.

--- a/App/BUCK
+++ b/App/BUCK
@@ -29,13 +29,13 @@ first_party_library_dependencies = [
     "//Libraries/SwiftWithMLModel:SwiftWithMLModel",
 ]
 
-apple_binary(
-    name = "ExampleAppBinary",
+apple_library(
+    name = "ExampleApp",
     visibility = [
         "//App:",
         "//App/...",
     ],
-    configs = binary_configs("ExampleApp"),
+    configs = library_configs(),
     swift_version = "4.0",
     srcs = [
         "ViewController.swift",
@@ -48,6 +48,23 @@ apple_binary(
         ":ExampleAppAssets",
     ] + first_party_library_dependencies,
 )
+
+apple_binary(
+    name = "ExampleAppBinary",
+    visibility = [
+        "//App:",
+        "//App/...",
+    ],
+    configs = binary_configs("ExampleApp"),
+    swift_version = "4.0",
+    srcs = [
+        "BuckSupportFiles/Dummy.swift",
+    ],
+    deps = [
+        ":ExampleApp",
+    ],
+)
+
 
 # This test bundles all unit test libraries into a single test target.
 # Test targets can be slow to create in CI; creating only one can save significant time.

--- a/App/BuckSupportFiles/Dummy.swift
+++ b/App/BuckSupportFiles/Dummy.swift
@@ -1,0 +1,3 @@
+// This file is meant to be included in an apple_binary that otherwise wouldn't have any code in it.
+// Buck apple_binary declarations need at least one source file in them to be properly generated.
+// Removing this file may make buck-generated Xcode projects fail to build.

--- a/App/Tests/BUCK
+++ b/App/Tests/BUCK
@@ -7,6 +7,6 @@ apple_test_lib(
     ]),
     info_plist = "Info.plist",
     deps = [
-        "//Libraries/Swift2:Swift2",
+        "//App:ExampleApp",
     ],
 )

--- a/App/Tests/BuckSampleTests.swift
+++ b/App/Tests/BuckSampleTests.swift
@@ -6,13 +6,14 @@
 //
 
 import XCTest
-@testable import Swift2
+@testable import ExampleApp
 
 final class BuckSampleTests: XCTestCase {
 
-    func testMySwiftClassAdding() {
-        let sut = MySwiftClass()
-        XCTAssertEqual(sut.add(1, 2), 3)
+    func testViewController() {
+        // This test accesses internal classes and properties in the ExampleApp library.
+        let sut = ViewController()
+        XCTAssertEqual(sut.value, "ViewName")
     }
 
     func testExample() {

--- a/Libraries/Swift2/Tests/SomeSwift2Tests.swift
+++ b/Libraries/Swift2/Tests/SomeSwift2Tests.swift
@@ -1,9 +1,11 @@
 import XCTest
+@testable import Swift2
 
 final class SomeSwift2Tests: XCTestCase {
 
-  func test_nothingAtAll() {
-    XCTAssertTrue(true)
+  func testMySwiftClassAdding() {
+    let sut = MySwiftClass()
+    XCTAssertEqual(sut.add(1, 2), 3)
   }
 
 }

